gfx_scene.h

```diff
@@ -1738,204 +1737,218 @@ class GfxSceneInternal
-                cgltf_accessor const *uv_buffer = it != gltf_primitive_attributes_end ? it->data : nullptr;
-                it = std::find_if(gltf_primitive.attributes, gltf_primitive_attributes_end,
-                    [&](const cgltf_attribute& x) { return x.type == cgltf_attribute_type_joints; });  // joints stream
-                cgltf_accessor const *joints_buffer = it != gltf_primitive_attributes_end ? it->data : nullptr;
-                it = std::find_if(gltf_primitive.attributes, gltf_primitive_attributes_end,
-                    [&](const cgltf_attribute& x) { return x.type == cgltf_attribute_type_weights; });  // weights stream
-                cgltf_accessor const *weights_buffer = it != gltf_primitive_attributes_end ? it->data : nullptr;
                 if(position_buffer == nullptr) continue; // invalid mesh primitive
-                std::vector<cgltf_float> sparse_position_buffer;
-                std::vector<cgltf_float> sparse_normal_buffer;
-                std::vector<cgltf_float> sparse_uv_buffer;
-                std::vector<cgltf_float> sparse_joints_buffer;
-                std::vector<cgltf_float> sparse_weights_buffer;
-                std::vector<cgltf_float> sparse_index_buffer;
-                auto unpack_sparse = [](cgltf_accessor const *buffer,
-                                        std::vector<cgltf_float> &sparse_buffer) -> bool {
-                    if(buffer != nullptr && buffer->is_sparse)
-                    {
-                        cgltf_size buffer_size = cgltf_num_components(buffer->type) * buffer->count;
-                        sparse_buffer.resize(buffer_size);
-                        if(cgltf_accessor_unpack_floats(buffer, &sparse_buffer[0], buffer_size) < buffer_size)
-                        {
-                            GFX_PRINT_ERROR(kGfxResult_InternalError, "Failed to unpack sparse accessor");
-                            return false;
-                        }
-                    }
-                    return true;
-                };
-                bool failed_unpack = !unpack_sparse(position_buffer, sparse_position_buffer);
-                failed_unpack      = failed_unpack || !unpack_sparse(normal_buffer, sparse_normal_buffer);
-                failed_unpack      = failed_unpack || !unpack_sparse(uv_buffer, sparse_uv_buffer);
-                failed_unpack      = failed_unpack || !unpack_sparse(joints_buffer, sparse_joints_buffer);
-                failed_unpack      = failed_unpack || !unpack_sparse(weights_buffer, sparse_weights_buffer);
-                failed_unpack      = failed_unpack || !unpack_sparse(index_buffer, sparse_index_buffer);
-                if(failed_unpack)
+                std::map<cgltf_accessor const *, GfxConstRef<GfxMesh>>::const_iterator mesh_it = meshInstances.find(position_buffer);//Note: assumes primitives with same position buffer will always have same attributes
+                if(mesh_it == meshInstances.end())
```

Comment:
This uses the position index as the mesh primitive key (some scenes have meshes with same indices but different positions so use positions). In the GLTF spec it is in theory possible for multiple meshes to have the same position index but differences in the other attributes (i.e. different vertex normals) but this should never actually happen